gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1121 @@
+    return
+
+def compute_post_variance(word, sslm, chain_variance):
+
+    T = sslm.num_sequences
+    variance = sslm.variance[word] # pick wordth row
+    fwd_variance = sslm.fwd_variance[word] # pick wordth row
+
+    # forward pass. Set initial variance very high
+    fwd_variance[0] = chain_variance * 1000
+    
+    for t in range(1, T + 1):
+        if sslm.obs_variance:
+            w = sslm.obs_variance / (fwd_variance[t - 1] + chain_variance + sslm.obs_variance)
+        else:
+            w = 0
+        fwd_variance[t] = w * (fwd_variance[t-1] + chain_variance)
+
+    # backward pass 
+    variance[T] = fwd_variance[T]
+    for t in range(T - 1, -1, -1):
+        if fwd_variance[t] > 0.0:
+            w = numpy.power((fwd_variance[t] / (fwd_variance[t] + chain_variance)), 2)
+        else:
+            w = 0
+        variance[t] = (w * (variance[t + 1] - chain_variance)) + ((1 - w) * fwd_variance[t])
+
+    sslm.variance[word] = variance
+    sslm.fwd_variance[word] = fwd_variance
+    return
+    
+
+def compute_post_mean(word, sslm, chain_variance):
+
+    T = sslm.num_sequences
+    obs = sslm.obs[word] # wordth row
+    mean = sslm.mean[word]
+    fwd_mean = sslm.fwd_mean[word]
+    fwd_variance = sslm.fwd_variance[word]
+
+    # forward 
+    fwd_mean[0] = 0
+    for t in range(1, T + 1):
+        w = sslm.obs_variance / (fwd_variance[t - 1] + chain_variance + sslm.obs_variance)
+        fwd_mean[t] = w * fwd_mean[t - 1] + (1 - w) * obs[t - 1]
+        if fwd_mean[t] is None:
+            # error message
+            pass
+
+    # backward pass
+    mean[T] = fwd_mean[T]
+    for t in range(T - 1, -1, -1):
+        if chain_variance == 0.0:
+            w = 0.0
+        else:
+            w = chain_variance / (fwd_variance[t] + chain_variance)
+        mean[t] = w * fwd_mean[t] + (1 - w) * mean[t + 1]
+        if mean[t] is None:
+            # error message
+            pass
+ 
+    return
+
+def compute_expected_log_prob(sslm):
+
+    W = sslm.num_terms
+    T = sslm.num_sequences
+    for t in range(0, T):
+        for w in range(0, W):
+            sslm.e_log_prob[w][t] = sslm.mean[w][t + 1] - numpy.log(sslm.zeta[t])
+    return
+
+
+def sslm_counts_init(sslm, obs_variance, chain_variance, sstats):
+
+    W = sslm.num_terms
+    T = sslm.num_sequences
+
+    log_norm_counts = numpy.copy(sstats)
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+
+    log_norm_counts = log_norm_counts + 1.0 / W
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+    log_norm_counts = numpy.log(log_norm_counts)
+    
+
+    # setting variational observations to transformed counts
+    for t in range(0, T):
+        sslm.obs[:,t] = log_norm_counts
+
+    # set variational parameters
+    sslm.obs_variance = obs_variance
+    sslm.chain_variance = chain_variance
+
+    # compute post variance
+    for w in range(0, W):
+       compute_post_variance(w, sslm, sslm.chain_variance)
```

Comment:
better to show explicitly the changes to sslm, no side effects. see comment in `compute_post_variance` above